### DISCUSSION TITLE
Error 42704 - referencing same table in column

### DISCUSSION THREAD

**Discussion by R-Mull** [supabase/supabase#20230] [Questions]
I'm using the table editor to add a column to a table 'staff' but I keep getting a 42704 error.
The new column should reference the same table 'staff'.
Basically each 'staff' member will have a supervisor who is another 'staff' member.

![Screenshot_20240108-160649](https://github.com/supabase/supabase/assets/154046331/1fafa316-04fd-4b50-a098-fec4aecd33aa)

Am I missing something obvious that's causing this?

**Comment by GaryAustin1** [comment]
If you have a compound primary key that is likely the issue if you are using just one of the columns for the foreign key.  Probably need to do the key set up with SQL.
